src/error.rs

```diff
@@ -95,8 +98,15 @@ pub enum StreamError {
     /// The first string is an actual one, others - expected.
     ///
     /// We are using a single value to reduce the struct size.
+    #[cfg(not(feature = "no_std"))]
     InvalidString(Vec<String>, TextPos),
 
+    /// An unexpected string.
+    ///
+    /// The byte string is the expected one.
+    #[cfg(feature = "no_std")]
+    InvalidString(&'static [u8], TextPos),
```

Comment:
This is ugly and I would've liked to add a reference to the actual string, but that would end up propagating life times to the `Error` crate, which would become a big pain to handle between having the `no_std` feature enabled and disabled. I would propose considering something like that as a breaking API change for the next major release.

Reply:
I guess we can make a breaking change here.